File: 1/prob01b.exs
defmodule Prob01b do
  def sum(file) do
    file
      |> Enum.map(&Integer.parse/1)
      |> Enum.map(&fuel/1)
      |> Enum.sum
  end

  def fuel({m, c}) do
    s = m / 3 - 2
      |> Kernel.trunc
    if (s > 0) do
      s + fuel({s, c})
    else
      0
    end
  end
end

File.stream!("input.txt", [:read])
  |> Prob01b.sum
  |> IO.puts

File: 1/prob01b.java
import java.io.*;

public class prob01b {
  public static int fuel(int m) {
    int s = m / 3 - 2;
    if (s > 0) {
      s += fuel(s);
      return s;
    } else {
      return 0;
    }
  }

  public static void main(String[] args) {
    try {
      FileReader in = new FileReader("input.txt");

      int sum = 0;
      String buffer = "";
      int c;
      while (true) {
        c = in.read();
        if (((char) c) == '\n' || c == -1) {
          sum += fuel(Integer.parseInt(buffer.trim()));
          buffer = "";
        } else {
          buffer += (char) c;
        }

        if (c == -1)
          break;
      }

      System.out.println(sum);
      in.close();
    } catch (IOException fe) {
      System.out.println("File not found!");
    }
  }
}
